io-engine/src/bdev/nexus/nexus_channel.rs

```diff
@@ -108,6 +110,7 @@ impl<'n> NexusChannel<'n> {
             io_mode: IoMode::Normal,
             frozen_ios: Vec::new(),
             core: Cores::current(),
+            is_io_chan: Thread::current().unwrap() != Thread::primary(),
```

Comment:
Might be frowned upon, but this kind of unwrap usage exists in io-engine core code today as well, and it is guaranteed that this code is running in spdk thread context.

Reply:
I think we should be comparing it the other way around, it's an io_chan if it's an nvmf thread? This is worth commenting out why we're doing this as in the future we might have a need to do IO from nexus channels.

But what bothers me the most is that we still have sync connect nonetheless... above on L88. Perhaps we should also avoid setting up readers and writers?

Reply:
> I think we should be comparing it the other way around, it's an io_chan if it's an nvmf thread? 

The current vs primary based check looks more finegrained. When you say nvmf thread - I guess it means there is a poll group attached to it for IO. nvmf target object itself doesn't seem to be spdk_thread bound. It's a thread_local (system thread local). So not sure if there is a clear way to identify this way.

> This is worth commenting out why we're doing this as in the future we might have a need to do IO from nexus channels.

Yeah I have a long code comment written, missed to paste here.

> But what bothers me the most is that we still have sync connect nonetheless... above on L88. Perhaps we should also avoid setting up readers and writers?

True. Things are in our favour here from the point that the probability of race is almost negligible because usually it's the reconfigurations that happen frequently and not this new channel construction. However, I agree that keeping the nexus channel dummy on init thread by not configuring `readers/writers` will be ok. The rebuild path anyway directly calls write api and not via `writers`. To achieve that, I'll move the check from L113 to very beginning, and do the connect loops only if bool is `true`.

Reply:
Isn't it this? https://github.com/openebs/mayastor/blob/558cfe7f798ce3de09552f224736133ea85024db/io-engine/src/subsys/nvmf/target.rs#L191

Reply:
> Isn't it this?

Yes it's this one. The way I can think using this info is something like below pseudocode:


```
let is_io_channel = false;
 t = current_spdk_thread();

loop_through_all_poll_groups_in_NVMF_PGS{
  If t_is_found_linked_with_some_poll_group (
   is_io_channel = true;
  )
}

```

Reply:
@tiagolobocastro  I'd say let's keep it as it's now in the PR, which is pretty reliable.

Reply:
Seems still not entirely clean, but if you're happy with the testing then it works for now.